xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-ui/src/main/resources/XWiki/Ratings.xml

```diff
@@ -394,9 +394,10 @@ position: relative;
     <className>XWiki.UIExtensionClass</className>
     <guid>d2420ff0-a5b9-4fae-a798-fc623aac7a6b</guid>
     <property>
-      <content>{{include reference='XWiki.RatingsMacros'/}}
-
-{{velocity}}
+      <content>{{velocity}}
+#if (!$isGuest)
```

Comment:
Well, yes, as you said. The issue is with the view right.
You don't really care if the user is a guest or logged in, you just want the check beforehand that the current user has viewing right on the macros page and if he doesn't I guess you could skip the whole logic described in the UI extension.
If memory serves me right, it would be something in the lines of:
#set($user = $xcontext.getUser())
#set($canView = $xwiki.getDocument('XWiki.RatingsMacros').hasAccessLevel('view', $user))
#if($canView)
## do stuff
#end